internal/app/machined/revert.go

```diff
@@ -0,0 +1,114 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package main
+
+import (
+	"context"
+	"log"
+	"os"
+
+	"github.com/siderolabs/go-blockdevice/blockdevice/probe"
+
+	"github.com/siderolabs/talos/internal/app/machined/pkg/runtime/v1alpha1/bootloader/grub"
+	"github.com/siderolabs/talos/internal/pkg/meta"
+	"github.com/siderolabs/talos/internal/pkg/mount"
+	"github.com/siderolabs/talos/pkg/machinery/constants"
+)
+
+func revertBootloader() {
+	if err := revertBootloadInternal(); err != nil {
+		log.Printf("failed to revert bootloader: %s", err)
+	}
+}
+
+//nolint:gocyclo
+func revertBootloadInternal() error {
+	metaState, err := meta.New(context.Background(), nil)
+	if os.IsNotExist(err) {
```

Comment:
```suggestion
	if !os.IsNotExist(err) {
		return err
	}
```

isn't this easier?

Reply:
this also works, but needs a check for `err != nil`, as `nil` will also satisfy the `!os.IsNotExist(err)` condition

Reply:
and here it is a bit different - we want to skip the revert if the META is not found, while we still return an error if we fail to read META

Reply:
> nil will also satisfy the !os.IsNotExist(err) condition

interesting, TIL

Reply:
I changed the code a bit to make it easier to read